rss2email/feeds.py

```diff
@@ -337,32 +339,37 @@ def save(self):
         dst_config_file = _os.path.realpath(self.configfiles[-1])
         _LOG.debug('save feed configuration to {}'.format(dst_config_file))
         for feed in self:
+            # Saves feed config to the config object, not the actual file (yet)
             feed.save_to_config()
         dirname = _os.path.dirname(dst_config_file)
         if dirname and not _os.path.isdir(dirname):
             _os.makedirs(dirname, mode=0o700, exist_ok=True)
-        tmpfile = dst_config_file + '.tmp'
-        with open(tmpfile, 'w') as f:
+
+        with open(dst_config_file, 'w') as f:
+            _fcntl.lockf(f, _fcntl.LOCK_EX) # lock for writing
             self.config.write(f)
             f.flush()
             _os.fsync(f.fileno())
-        _os.replace(tmpfile, dst_config_file)
```

Comment:
i think that for the config file, the write-then-mv behavior should be preferred, as `mv` is (usually) atomic, while write is not -- this reduces the risk of losing the configuration. Plus, if multiple processes are attempting configuration change at the same time, something is clearly wrong, and if there is a race condition it's not that bad.